cg/src/domains/subjects/subjects.service.js

```diff
@@ -33,18 +35,21 @@ class SubjectsService {
 
     if (subjectExists) throw new Unauthorized('Subject already gave consent');
     let processorIdsWithAddresses;
-    await this.db.transaction(async trx => {
-      await this._initializeUserInTransaction(trx, subjectId, personalData);
-      processorIdsWithAddresses = await this._getProcessorIdsWithAddresses(
-        trx,
-        processorsConsented
-      );
-      await this._verifyProcessors(processorIdsWithAddresses, processorsConsented);
-      await Promise.all(
-        processorsConsented.map(processor => this._setConsentGiven(trx, subjectId, processor))
-      );
+    
+    return this.db.transaction(async trx => {
+      await trx.transaction(async (nestedTrx) => {
```

Comment:
That's because we can have a race condition between the processor listening to events and the controlle rdatabase with only 1 transaction. We need some sort of checkpoint between the database insertion and the smart-contract event call
A nested transaction works, but there may be more ways to do it